### PR TITLE
Async Functions Sample for Node v4.0.0

### PR DESCRIPTION
**Pull request by rbuckton**
Sample of Async Functions using NodeJS v4.0.

### CODE REVIEW

File: async/jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (mhegazy) [inline async/jsconfig.json]: What is this file used for?

File: async/src/lib/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "compilerOptions": {
+        "target": "ES6",
+        "module": "commonjs",
+        "experimentalAsyncFunctions": true
+    },
+    "files": [
```
Comment (mhegazy) [inline async/src/lib/tsconfig.json]: Why not leave it with no "files" property? makes it user for ppl to play with the sample and add new files..

**Comment by mhegazy** [comment]
:+1:

File: async/jsconfig.json (hunk 1 above)
Comment (rbuckton) [inline async/jsconfig.json]: Gets rid of errors when viewing the generated JavaScript in VS Code.

File: async/src/lib/tsconfig.json (hunk 2 above)
Comment (rbuckton) [inline async/src/lib/tsconfig.json]: Sure.

**Pull request merged**